scripts/common

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env bash
```

Comment:
All this bash script logic makes me slightly uneasy, I would definitely not trust myself to get it right or even maintain it, but it's fine if you feel that it is working correctly. Should we rewrite it in Rust perhaps? have a small binary that orchestrates tests (mostly by calling out to external scripts or docker commands, granted)?